lib/active_merchant/billing/gateways/netbanx.rb

```diff
@@ -157,8 +157,8 @@ def expdate(credit_card)
         { :month => month, :year => year }
       end
 
-      def add_order_id(post, options)
-        post[:merchantRefNum] = (options[:order_id] || SecureRandom.uuid)
+      def add_order_id(post)
+        post[:merchantRefNum] = (SecureRandom.uuid)
```

Comment:
You don't need the parenthesis here.

Reply:
The reason for us to make this change is that the PaySafe gateway expects a unique merchantRefNum each time we make a partial refund for an order. In the code the order id was being sent as the merchantRefNum for each refund and this allowed only 1 refund per order (full or partial).

Given that the earlier code change affected multiple line, we have now moved the only change we wanted into its own method. We just needed this functionality in the refunds method, not anywhere else. 

A test case won't be required now because this line of change will only affect the refunds method and as per the existing test cases it's not possible to test the refunds without going through the full transaction and settlement cycle from the gateway.

With this final commit we hope that we can push this change onto the Shopify platform and it won't have any negative affect on any functionality except refunds.

Additionally all existing test cases are being satisfied after this code change and also the automatic build on our branch is passing.